File: 102_二叉树的层序遍历/二叉树的层序遍历.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        res = []
        if not root:
            return res
        dq = deque([root])
        while dq:
            n = len(dq)
            level = []
            for _ in range(n):
                cur = dq.popleft()
                level.append(cur.val)
                if cur.left:
                    dq.append(cur.left)
                if cur.right:
                    dq.append(cur.right)
            res.append(level)
        return res

File: 102_二叉树的层序遍历/二叉树的层序遍历.js
/**
 * @param {TreeNode} root
 * @return {number[][]}
 */
var levelOrder = function (root) {
    ans = []
    if (!root) {
        return ans
    } else {
        const deque = [root]
        while (deque.length != 0) {
            level = []
            len = deque.length
            for (var i = 0; i < len; i++) {
                var cur = deque.shift()
                level.push(cur.val)
                if (cur.left) {
                    deque.push(cur.left)
                }
                if (cur.right) {
                    deque.push(cur.right)
                }
            }
            ans.push(level)
        }
        return ans
    }
};